# Experiment 7 - Text Analysis

In [40]:
import pandas as pd 
import numpy as np

### Problem 1.1 – Loading the Dataset

In [41]:
df = pd.read_csv('emails.csv')

In [42]:
df.head()

text  spam
0  Subject: naturally irresistible your corporate...     1
1  Subject: the stock trading gunslinger  fanny i...     1
2  Subject: unbelievable new homes made easy  im ...     1
3  Subject: 4 color printing special  request add...     1
4  Subject: do not have money , get software cds ...     1

In [43]:
df.describe()

spam
count  5728.000000
mean      0.238827
std       0.426404
min       0.000000
25%       0.000000
50%       0.000000
75%       0.000000
max       1.000000

In [44]:
df.shape

(5728, 2)

In [45]:
spam_count = np.sum(np.asarray(df.spam) ) 
spam_count

1368

In [46]:
# Check of 1st word:
max_length = 0
for item in np.asarray(df.text):
    max_length = max(len(item), max_length)
    texts = item.split(":")
    if texts[0] != "Subject": 
        print("Subject is not the First word")
        break 

print("Maximum length of the longest text is: ", max_length)
    

Maximum length of the longest text is:  43952


### Answers of Problem 1

* How many emails are in the dataset?<br>
-> There are 5728 emails in the Dataset.


* How many of the emails are spam?<br>
-> Count of Spam emails = 1368

* Which word appears at the beginning of every email in the dataset? <br>
"Subject" is the first word in all the texts. <br>

* Respond as a lower-case word with punctuation removed. <br>
"subject"

* Could a spam classifier potentially benefit from including the frequency of the word that appears in every email?<br>
-> Yes -- the number of times the word appears might help us differentiate spam from ham.

* The nchar() function counts the number of characters in a piece of text. <br>
How many characters are in the longest email in the dataset (where longest is measured in terms of the maximum number of characters)?<br>
-> 43952


### Problem 2.1 -Preparing the Corpus

In [47]:
corpus = df.copy()

In [48]:
corpus['text'] = corpus['text'].str.replace('[^a-zA-Z#]', " ")
corpus

text  spam
0     Subject  naturally irresistible your corporate...     1
1     Subject  the stock trading gunslinger  fanny i...     1
2     Subject  unbelievable new homes made easy  im ...     1
3     Subject    color printing special  request add...     1
4     Subject  do not have money   get software cds ...     1
...                                                 ...   ...
5723  Subject  re   research and development charges...     0
5724  Subject  re   receipts from visit  jim    than...     0
5725  Subject  re   enron case study update  wow   a...     0
5726  Subject  re   interest  david    please   call...     0
5727  Subject  news   aurora       update  aurora ve...     0

[5728 rows x 2 columns]

In [49]:
stop_words = ["","i", "me", "my", "myself", "we", "our", "ours", "ourselves", "you", "your", "yours", "yourself", "yourselves", "he", "him", "his", "himself", "she", "her", "hers", "herself", "it", "its", "itself", "they", "them", "their", "theirs", "themselves", "what", "which", "who", "whom", "this", "that", "these", "those", "am", "is", "are", "was", "were", "be", "been", "being", "have", "has", "had", "having", "do", "does", "did", "doing", "would", "should", "could", "ought", "i'm", "you're", "he's", "she's", "it's", "we're", "they're", "i've", "you've", "we've", "they've", "i'd", "you'd", "he'd", "she'd", "we'd", "they'd", "i'll", "you'll", "he'll", "she'll", "we'll", "they'll", "isn't", "aren't", "wasn't", "weren't", "hasn't", "haven't", "hadn't", "doesn't", "don't", "didn't", "won't", "wouldn't", "shan't", "shouldn't", "can't", "cannot", "couldn't", "mustn't", "let's", "that's", "who's", "what's", "here's", "there's", "when's", "where's", "why's", "how's", "a", "an", "the", "and", "but", "if", "or", "because", "as", "until", "while", "of", "at", "by", "for", "with", "about", "against", "between", "into", "through", "during", "before", "after", "above", "below", "to", "from", "up", "down", "in", "out", "on", "off", "over", "under", "again", "further", "then", "once", "here", "there", "when", "where", "why", "how", "all", "any", "both", "each", "few", "more", "most", "other", "some", "such", "no", "nor", "not", "only", "own", "same", "so", "than", "too", "very"]
corpus_text = np.asarray(corpus.text)

In [50]:
for i in range(len(corpus_text)): 
    corpus_text[i] = corpus_text[i].lower().split()
    print(len(corpus_text[i]))
    corpus_text[i].remove(stop_words)
    print(len(corpus_text))
    break
    

188


ValueError: list.remove(x): x not in list